networking/podenv.go

```diff
@@ -154,6 +173,25 @@ func (e *podEnv) podNSPathSave() error {
 	return nil
 }
 
+func (e *podEnv) podNSDestroy() error {
+	if e.podNS == nil {
+		return nil
+	}
+	e.podNS.Close() // Don't care if err here
+
+	if err := syscall.Unmount(e.podNS.Path(), unix.MNT_DETACH); err != nil {
+		// if already unmounted, umount(2) returns EINVAL
+		if !os.IsNotExist(err) && err != syscall.EINVAL {
+			return errwrap.Wrap(fmt.Errorf("error unmounting %q", e.podNS.Path()), err)
+		}
+	}
+
+	if err := os.RemoveAll(e.podNS.Path()); err != nil {
```

Comment:
I thought about this, and I decided we shouldn't - trying to remove a mounted mountpoint will fail.